### ISSUE TITLE
New lint to remove `??` inside `if null` test when the body does more things

### ISSUE THREAD

**Issue by FMorschel** [dart-lang/linter#5109]
Repro:

```dart
class A {
  int? value;
  void foo() {
    if (value == null) {
      value ??= 0;
    }
  }
}
``` 

This is a code that triggers `prefer_conditional_assignment` with a quick-fix that suggests removing the `if`.

```dart
class A {
  int? value;
  void foo() {
    if (value == null) {
      value ??= 0;
      print(value);
    }
  }
}
``` 

This code is the same but it does more things, so in this case I believe we should have a new lint + quick-fix for reminding the user to remove the `??` from the assignment expression since it doesn't do anything. 

As the bot reminded me to be explicit: the second code has **_no lint_** today.

**Comment by srawlins** [comment]
Yeah not a bad idea. I think we need information about whether a promotable variable is _definitely_ `null` (or `Null` type). But I think currently we don't promote on `== null`. I think there is a language issue for this proposal.

**Comment by FMorschel** [comment]
Do you mean this https://github.com/dart-lang/language/issues/3031?

**Comment by srawlins** [comment]
This one: https://github.com/dart-lang/language/issues/1505
